fields/src/macros.rs

```diff
@@ -273,6 +274,10 @@ macro_rules! impl_primefield_serializer {
 
                 let mut masked_bytes = [0; $byte_size + 1];
                 reader.read_exact(&mut masked_bytes[..output_byte_size])?;
+                let formatted_string = format!("{:?} , {}", masked_bytes, output_byte_size);
```

Comment:
Super weird bug here. It deserializes differently from the wasm vs the rust test. Not sure why yet, maybe some deserialization reordering. This happens when deserializing the transfer prover file.

This is the bug preventing it from working. 

<img width="1674" alt="Screenshot 2023-02-06 at 10 32 06 PM" src="https://user-images.githubusercontent.com/1102811/217168955-d41a0167-a298-49ae-9f5a-02658bf11c90.png">
<img width="1319" alt="Screenshot 2023-02-06 at 10 32 39 PM" src="https://user-images.githubusercontent.com/1102811/217168925-456f89b6-4ebf-49c9-bc9a-af628c487a62.png">


<img width="1653" alt="Screenshot 2023-02-06 at 10 33 16 PM" src="https://user-images.githubusercontent.com/1102811/217168896-3f9303e4-5364-4ea6-a3e0-25905b4711cf.png">